presto-main/src/main/java/com/facebook/presto/dispatcher/DispatchManager.java

```diff
@@ -187,11 +188,11 @@ private <C> void createQueryInternal(QueryId queryId, String slug, int retryCoun
                 throw new PrestoException(QUERY_TEXT_TOO_LARGE, format("Query text length (%s) exceeds the maximum length (%s)", queryLength, maxQueryLength));
             }
 
-            // check permissions
-            checkPermissions(queryId, sessionContext);
+            // get authorized identity if possible
+            Optional<AuthorizedIdentity> authorizedIdentity = getAuthorizedIdentity(queryId, sessionContext);
```

Comment:
I would recommend split the current method checkPermissions into 2 methods. 

checkPermissions, which checks permission when the new config is not enabled.
getAuthorizedIdentity when the config is enabled. 

Modify the cod here to something like this.
Optional<AuthorizedIdentity> optionalField;
if (new config not enabled) {
     checkPermissions;
     optionalField = empty;
}
else {
    optionalField = getAuthorizedIdentity
}

The current code is very hacky with inconsistent name, and does permission check inside.

Reply:
I worried about this too (single responsibility). However, since getAuthorizedIdentity also does delegation check internally, these two functionalities are somehow similar, so I put them together. But your confusion is a good evidence for better readability.

Reply:
Done